codegen/src/main/ops/org/nd4j/codegen/ops/SDBaseOps.kt

```diff
@@ -0,0 +1,1806 @@
+        Input(NUMERIC, "in") { description = "Input" }
+        Arg(CONDITION, "condition") { description = "Condition" }
+        Output(NUMERIC, "output"){ description = "Number of elements that the condition is satisfied for" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Returns a count of the number of elements that satisfy the condition
+            """.trimIndent()
+        }
+    }
+
+    Op("matchConditionCount") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.reduce.longer"
+        javaOpClass = "MatchCondition"
+        legacy = true
+        Input(NUMERIC, "in") { description = "Input variable" }
+        Arg(CONDITION, "condition") { description = "Condition" }
+        Arg(BOOL, "keepDim") { description = "If true: keep the dimensions that are reduced on (as size 1). False: remove the reduction dimensions" }
+        Arg(INT, "dimensions") {count = AtLeast(1); description = "Dimensions to reduce over. If dimensions are not specified, full array reduction is performed" }
+        Output(NUMERIC, "output"){ description = "Number of elements that the condition is satisfied for" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Returns a count of the number of elements that satisfy the condition (for each slice along the specified dimensions)
+                Note that if keepDims = true, the output variable has the same rank as the input variable,
+                with the reduced dimensions having size 1. This can be useful for later broadcast operations (such as subtracting
+                the mean along a dimension).
+                Example: if input has shape [a,b,c] and dimensions=[1] then output has shape:
+                keepDims = true: [a,1,c]<br>
+                keepDims = false: [a,c]
+            """.trimIndent()
+        }
+    }
+
+    Op("max") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.reduce.same"
+        legacy = true
+        Input(NUMERIC, "x") { description = "Input variable" }
+        Arg(INT, "dimensions") {count = AtLeast(1); description = "Dimensions to reduce over. If dimensions are not specified, full array reduction is performed" }
+        Output(NUMERIC, "output"){ description = "Reduced array of rank (input rank - num dimensions)" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Max array reduction operation, optionally along specified dimensions
+            """.trimIndent()
+        }
+    }
+
+    Op("max") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.reduce.same"
+        legacy = true
+        Input(NUMERIC, "x") { description = "Input variable" }
+        Arg(BOOL, "keepDims") { description = "If true: keep the dimensions that are reduced on (as size 1). False: remove the reduction dimensions" }
+        Arg(INT, "dimensions") { count = AtLeast(1); description = "Dimensions to reduce over. If dimensions are not specified, full array reduction is performed" }
```

Comment:
No dimensions is allowed -> full array reduction

Reply:
The comment on dimensions does say this.